src/player.js

```diff
@@ -271,16 +285,20 @@ export default class Player extends FakeEventTarget {
    * @public
    */
   destroy(): void {
-    if (this._engine) {
-      this._engine.destroy();
-    }
+    this._canLoadEnginePromise = this._engine ? this._engine.reset() : Promise.resolve();
```

Comment:
Why? do u see any use case for this?
If we already decide to do something like this, lets expose destroy() and reset() of the player as promises (this promise exists to avoid exactly this).